csharp/src/Google.Protobuf/Reflection/DescriptorPartial.cs

```diff
@@ -0,0 +1,50 @@
+namespace Google.Protobuf.Reflection
+{
+    public sealed partial class FileOptions 
+    {
+        internal ExtensionSet<FileOptions> Extensions => _extensions;
```

Comment:
Looks like these two properties are only needed to create an instance of custom options.
If so, why expose the properties at all and instead add a CreateCustomOptions() method which creates the CustomOptions instance for you.